tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -597,6 +630,141 @@ int main(int argc, char **argv)
+
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+
+    /* Test: s2n_select_resumption_psk */
+    {
+        /* Safety checks */
+        {
+            struct s2n_stuffer wire_identities_in = { 0 };
+            struct s2n_connection *conn = s2n_connection_new(S2N_SERVER);
+            EXPECT_NOT_NULL(conn);
+
+            EXPECT_ERROR_WITH_ERRNO(s2n_select_resumption_psk(NULL, &wire_identities_in), S2N_ERR_NULL);
+            EXPECT_ERROR_WITH_ERRNO(s2n_select_resumption_psk(conn, NULL), S2N_ERR_NULL);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+        }
+
+        /* Setup for creating a test resumption psk */
+        {
+            DEFER_CLEANUP(struct s2n_stuffer wire_identities_in = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&wire_identities_in, 0));
+
+            struct s2n_connection *conn = s2n_connection_new(S2N_SERVER);
+            EXPECT_NOT_NULL(conn);
+
+            struct s2n_config *config = s2n_config_new();
+            EXPECT_NOT_NULL(config);
+            EXPECT_SUCCESS(s2n_setup_ticket_key(config));
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+
+            conn->actual_protocol_version = S2N_TLS13;
+            conn->secure.cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+
+            DEFER_CLEANUP(struct s2n_stuffer psk_identity = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&psk_identity, 0));
+            EXPECT_OK(s2n_setup_encrypted_ticket(conn, &psk_identity));
+
+            /* One valid resumption psk is received */
+            {
+                EXPECT_OK(s2n_write_test_identity(&wire_identities_in, &psk_identity.blob));
+
+                EXPECT_OK(s2n_select_resumption_psk(conn, &wire_identities_in));
+
+                EXPECT_EQUAL(conn->psk_params.chosen_psk_wire_index, 0);
+                EXPECT_NOT_NULL(conn->psk_params.chosen_psk);
+
+                /* Sanity check psk creation is correct */
+                EXPECT_EQUAL(conn->psk_params.chosen_psk->hmac_alg, s2n_tls13_aes_128_gcm_sha256.prf_alg);
+            }
+
+            /* First valid resumption psk is chosen */
+            {
+                conn->psk_params.chosen_psk = NULL;
+                EXPECT_SUCCESS(s2n_stuffer_rewrite(&wire_identities_in));
+
+                /* Write invalid resumption psk */
+                uint8_t bad_identity_data[] = "hello";
+                struct s2n_blob bad_identity = { 0 };
+                EXPECT_SUCCESS(s2n_blob_init(&bad_identity, bad_identity_data, sizeof(bad_identity_data)));
+
+                uint8_t valid_psk_idx = 5;
+                for (size_t i = 0; i < valid_psk_idx; i++) {
+                    EXPECT_OK(s2n_write_test_identity(&wire_identities_in, &bad_identity));
+                }
+
+                /* Write valid resumption psk */
+                EXPECT_OK(s2n_write_test_identity(&wire_identities_in, &psk_identity.blob));
```

Comment:
Add more than one valid resumption psk to the list and verify that the first valid resumption psk is the chosen psk.